lib/cpp/src/thrift/transport/THeaderTransport.h

```diff
@@ -135,8 +141,9 @@ class THeaderTransport : public TVirtualTransport<THeaderTransport, TFramedTrans
   void transform(uint8_t* ptr, uint32_t sz);
 
   uint16_t getNumTransforms() const {
-    int trans = writeTrans_.size();
-    return trans;
+    std::vector<uint16_t>::size_type trans = writeTrans_.size();
+    assert(trans <= UINT16_MAX);
```

Comment:
If your casting error checking is just debug asserts, then you might as well stick with the casts.
My preference would be to throw an exception if the size is out of bounds.